JCEF attempt JBR

KlemenDEV opened the pull request:
Windows:

* **[non OSR] works without known issues** ✅
* [OSR] black flash at start ⚠️

macOS:

* **[non OSR] works without known issues** ✅
* [OSR] not working on aarch64 due to wrong natives ❌

Linux:

* [non OSR] big problems with focus transfer ❌
* **[OSR] works fine on Ubuntu** ✅

KlemenDEV committed:
JCEF attempt, but gave up for now (complex install, async JS would require a lot of refactoring - need to implement in-house sync solution?)

KlemenDEV committed:
Merge branch 'master' into jcef_attempt

# Conflicts:
#	build.gradle
#	src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java
#	src/main/java/net/mcreator/ui/blockly/WebConsoleListener.java

KlemenDEV committed:
Manual merge

KlemenDEV committed:
Some more untested work, build fails, updated JCEF version

KlemenDEV committed:
Minor refactor

KlemenDEV committed:
Changes

KlemenDEV committed:
Merge branch 'master' into jcef_attempt

KlemenDEV committed:
Changes

KlemenDEV committed:
Blockly loads, key events are passed, focus seems to work - only tested on Windows - non OSR, OSR mode closes whole app when closing tab - open TODOs present

KlemenDEV committed:
Improvements

KlemenDEV committed:
Correct handling of Swing dialogs via javascript bridge

KlemenDEV committed:
Preload browser

KlemenDEV committed:
One client per browser instead of shared client + made JS bridge completely backing system indepdendent

KlemenDEV committed:
Support for blocking sync calls

KlemenDEV committed:
Too hardcoded

KlemenDEV committed:
Removed transparency parameter, it will not work in JCEF

KlemenDEV committed:
Some hacks

KlemenDEV committed:
Some changes

KlemenDEV committed:
Merge branch 'master' into jcef_attempt

KlemenDEV committed:
Changes to dist setup to not download natives, but it still extracts them - added TODO

KlemenDEV committed:
Changes

KlemenDEV committed:
Some minor fixes

KlemenDEV committed:
No OSR

KlemenDEV committed:
Some JCEF bug workarounds

KlemenDEV committed:
In-memory cache

KlemenDEV committed:
Minor change

KlemenDEV committed:
Use per-webview thread so crash in one instance does not hang all of them

KlemenDEV committed:
Some work, focus not ok on non-OSR mode

KlemenDEV committed:
Bug fix

KlemenDEV committed:
Merge branch 'jcef_attempt' into jcef_attempt_jbr

KlemenDEV committed:
Working correctly using OSR mode on Windows

KlemenDEV committed:
JBR attempt

KlemenDEV committed:
Some attempts

KlemenDEV committed:
Did not fix

KlemenDEV committed:
More changes

NerdyPuzzle commented:
https://imgur.com/NL8PmZ5

there seems to be flickering on the cursor between regular icon and grabbing icon when moving it over procedure blocks or moving large procedures

KlemenDEV commented:
Does it also happen with the non-JBR version?

NerdyPuzzle commented:
nope!

KlemenDEV committed:
Missing files + load blockly CSS directly for better response

KlemenDEV committed:
Fix macOS hanging on placing blocks from search bug?

KlemenDEV committed:
Some fixes

KlemenDEV committed:
Unused stuff

KlemenDEV committed:
Different macOS and non-macOS handling

KlemenDEV committed:
More minor changes

KlemenDEV committed:
Make it compile on macOS

KlemenDEV committed:
Threading issues fixes on macOS?

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
Merge branch 'klemen-master' into jcef_attempt_jbr

KlemenDEV committed:
System file chooser using FlatLAF (#5970)

* Initial implementation

* Non ideal solution for armor selector

* FlatLAF 3.7 (non-snapshot version)

KlemenDEV committed:
Complete removal of JavaFX

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

NerdyPuzzle committed:
Implement get tick count of entity procedure block in order to help optimize tick loops (#5874)

* Implement tickcount procedure

* Update texts.properties

* Update texts.properties

* Update texts.properties

KlemenDEV committed:
JBR JDK?

KlemenDEV committed:
JBR JDK 2?

KlemenDEV committed:
Don't use OSR for tests

KlemenDEV committed:
Fix logging without javafx + fix dependency system for CEF JS bridge

KlemenDEV committed:
Updated some GitHub Actions

KlemenDEV committed:
Merge remote-tracking branch 'origin/master'

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
Fix dependency backwards compat

KlemenDEV committed:
No GPU on github actions?

KlemenDEV committed:
GH auth for JBR download to prevent rate limiter

KlemenDEV committed:
Texture size should be multiple of 16 for nicer rendering

KlemenDEV committed:
Manual merge

KlemenDEV committed:
Merge branch 'refs/heads/master' into jcef_attempt_jbr

KlemenDEV committed:
Support multiple JS bridges correctly

KlemenDEV committed:
Fix input stream leak

KlemenDEV committed:
Fixed some memory leaks

KlemenDEV committed:
Plugin loader resources cache to avoid RAM and CPU heavy lookups

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
Minor change

KlemenDEV committed:
Fix hanging threads bug with webview when closed

KlemenDEV committed:
Changes

KlemenDEV committed:
More changes for CI

KlemenDEV committed:
Some more flags

KlemenDEV committed:
One flag less

KlemenDEV committed:
Less flags

KlemenDEV committed:
Less flags

Goldorion committed:
"Any item in" block for advancements and related changes (Item conditions 1/3) (#5828)

* MCItemBlock's code is now the one generating ""

* Add Any items of + a new category

* Warn unknown type

* Small changes

CRLF = grr
Grammar = I'm still bad with English :(
Category description = :_(

* Fix item_enchanted not being a list of items

* Rename the files for the "Any item of" change

* Any item in is now always use for testing an item-related trigger

* ... Oops

It's better when you define what to replace

* Change tests + use fields for MCItemSelectors

* Use a JSON list only when appropriated

Follow the same logic as Block holderset list (custom features)

---------

Co-authored-by: KlemenDEV <16374228+KlemenDEV@users.noreply.github.com>

KlemenDEV committed:
More flags

KlemenDEV committed:
GC?

KlemenDEV committed:
Revert "GC?"

This reverts commit 057ec883c3e62d28ff1bfc334963a4d1c3cf834e.

KlemenDEV committed:
Fix typo

KlemenDEV committed:
Some new flags

KlemenDEV committed:
Less RAM usage

KlemenDEV committed:
Single process?

KlemenDEV committed:
Fix some native file dialogs bugs

KlemenDEV committed:
Merge remote-tracking branch 'origin/master'

KlemenDEV committed:
Dependency bump

KlemenDEV committed:
Fix generator stats showing partial support when they shouldn't

KlemenDEV committed:
Do not preload fallback cache for default theme

KlemenDEV committed:
A bit less test combinations where they are redundant

KlemenDEV committed:
No need to test file usages system anymore as now it does not rely on MET definitions anymore

KlemenDEV committed:
Removed some of old test workspaces for faster tests

KlemenDEV committed:
Revert "Single process?"

This reverts commit 142dbffb6ac8964e11d9d334e7a5e15c7f579662.

KlemenDEV committed:
Revert "Less RAM usage"

This reverts commit c04a9943b48161565517c08cf5522ed3ca5dde20.

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
Run UI tests for all METs + some typo fixes

Goldorion committed:
Add a hasShadow param for labels (#5869)

* Add a hasShadow param for labels

* Better shadow preview

KlemenDEV committed:
More RAM-friendly tests

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
Export scripts

KlemenDEV committed:
Correct JBR version

KlemenDEV committed:
Always non-OSR

KlemenDEV committed:
Well...

KlemenDEV committed:
New code signing TS URL

AlsoSomeoneElse committed:
"Flip image" tools for image editor (#5888)

* Add "Flip image" tools

* Add keyboard shortcuts (V and Shift+V)

* Remove additional "flipImage" method

* Do not create additional buffered images

KlemenDEV committed:
Do not exclude java base jmod

KlemenDEV committed:
Fix Gradle Java installation specification

KlemenDEV committed:
Fix toolchains resolving system that was not working on slower computers or certain Gradle setup

KlemenDEV committed:
Merge remote-tracking branch 'origin/master'

KlemenDEV committed:
Fix invalid JSON

KlemenDEV committed:
Log before failing tests

KlemenDEV committed:
Improved blockly test field detection

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
Fix windows key event conversion

KlemenDEV committed:
Do not call createImmediately outside preload and tests, as it breaks loading on windows in living entity editor tab

KlemenDEV committed:
Minor UI fix on Linux

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr

KlemenDEV committed:
OSR settings

NerdyPuzzle committed:
Add the option for armor models to support translucent textures (#5926)

* Implement GUI

* Implement 1.21.8 rendering

* Implement 1.21.1 rendering

* Forgot to implement tests

* Add warnings about performance implications

* Fix random whitespace before posestack

* Put the checkbox under the model selector

* Split the rendering to multiple lines

* add javacompress directive to 1.21.1

github-actions[bot] committed:
New Crowdin translations (#5978)

Co-authored-by: github-actions <>

KlemenDEV committed:
Fix #5977 (#5979)

KlemenDEV committed:
Merge branch 'master' into jcef_attempt_jbr
